operator/pkg/controller/clusterconfig/workloadplane/virtualippool.go

```diff
@@ -14,24 +14,37 @@ import (
 	"github.com/scality/metalk8s/operator/pkg/controller/utils"
 )
 
+type VirtualIPPoolReconciler struct {
+	handler *utils.ObjectHandler
+	logger  logr.Logger
+
+	instance *metalk8sscalitycomv1alpha1.ClusterConfig
```

Comment:
If each reconciler holds a reference to this instance, and they all run concurrently, won't we have data races? Seems fishy to me, wondering if they shouldn't rather receive a copy 😕

Reply:
Each reconciler should only "manage" his own part of the `status` so we shouldn't have any issue.

The issue about copying it is that each sub reconcile may edit the `status` and add the end (with `defer` from the "main" controller) we actually push the changes, so if we do a copy we will have to "reconcile" the `status` content and it might be a mess (especially if we consider that 2 different sub reconcile may edit the same status)

Reply:
So I added a Mutex logic to handle Conditions for ClusterConfig

See https://github.com/scality/metalk8s/pull/4000/commits/2fbab1f825034ac1e3c199008e010affe5468b0a